runtime/web_worker.rs

```diff
@@ -122,40 +116,45 @@ impl WebWorkerInternalHandle {
   }
 }
 
-#[derive(Clone)]
-pub struct WebWorkerHandle {
-  sender: mpsc::Sender<WorkerMessage>,
-  receiver: Arc<AsyncMutex<mpsc::Receiver<WorkerEvent>>>,
+pub struct SendableWebWorkerHandle {
+  port: MessagePort,
+  receiver: mpsc::Receiver<WorkerControlEvent>,
   terminated: Arc<AtomicBool>,
   isolate_handle: v8::IsolateHandle,
 }
 
-impl WebWorkerHandle {
-  /// Post WorkerMessage to worker as a host
-  pub fn post_message(&self, buf: WorkerMessage) -> Result<(), AnyError> {
-    let mut sender = self.sender.clone();
-    // If the channel is closed,
-    // the worker must have terminated but the termination message has not yet been recieved.
-    //
-    // Therefore just treat it as if the worker has terminated and return.
-    if sender.is_closed() {
-      self.terminated.store(true, Ordering::SeqCst);
-      return Ok(());
+impl From<SendableWebWorkerHandle> for WebWorkerHandle {
+  fn from(handle: SendableWebWorkerHandle) -> Self {
+    WebWorkerHandle {
+      receiver: Rc::new(RefCell::new(handle.receiver)),
+      port: Rc::new(handle.port),
+      terminated: handle.terminated,
+      isolate_handle: handle.isolate_handle,
     }
-    sender.try_send(buf)?;
-    Ok(())
   }
+}
 
+#[derive(Clone)]
+pub struct WebWorkerHandle {
+  pub port: Rc<MessagePort>,
+  receiver: Rc<RefCell<mpsc::Receiver<WorkerControlEvent>>>,
+  terminated: Arc<AtomicBool>,
+  isolate_handle: v8::IsolateHandle,
+}
```

Comment:
I'm I little confused, why do we need three different handles?

Reply:
One is Send, the other one is Clone. Having the same struct do both requires mutexes, which were complicating the code and were causing various issues.

Reply:
The `SendableWebWorkerHandle` is transferred from the worker thread to the parent thread, where it is turned into a regular `WebWorkerHandle`.

Reply:
Okay sounds good, please update docstrings for those structs and it's good for me to land